android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/KeymanSettingsActivity.java

```diff
@@ -22,5 +24,16 @@ protected void onCreate(Bundle savedInstanceState) {
       getSupportActionBar().setDisplayShowHomeEnabled(true);
       getSupportActionBar().setDisplayShowTitleEnabled(true);
     }
+
+    innerFragment = (KeymanSettingsFragment) getSupportFragmentManager().findFragmentById(R.id.keyman_settings_fragment);
+  }
+
+  @Override
+  public void onWindowFocusChanged(boolean hasFocus) {
+    super.onWindowFocusChanged(hasFocus);
+
+    if(hasFocus) {
+      innerFragment.update();
```

Comment:
This causes the preference checkboxes to update after a child activity has been opened, correct?